python/test/unit/fem/test_assemble_submesh.py

```diff
@@ -13,8 +13,9 @@
 
 import ufl
 from dolfinx import default_scalar_type, fem, la
+from dolfinx.fem.petsc import assemble_matrix
```

Comment:
I think that @chrisrichardson was trying to untie these tests from PETSc. Discuss with him whether assembly in the builtin matrix format would be acceptable here, or not.